### PR TITLE
fix(core): allow functions in DeletionPolicy and UpdateReplacePolicy

### PR DESCRIPTION
**Pull request by bertrandda**
With [`AWS::LanguageExtensions`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-languageextension-transform.html) it's possible to use functions in `DeletionPolicy` and `UpdateReplacePolicy`.

When we try to include a template with a function in DeletionPolicy attribut (or UpdateReplacePolicy) in CDK stack, it fails with error
```
Error: Unrecognized DeletionPolicy '[object Object]'
```

This PR add a call to function parser when parsed cloudformation template contains DeletionPolicy as object.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by bertrandda** [comment]
`Exemption Request`
Changes are only on a helpers function, I added related unit test but I don't think a integration test is necessary.

**Review by corymhall — CHANGES_REQUESTED** [review]
@bertrandda in order to use language extensions don't you need to include the transform in the template?

Can you also add an integration test to make sure that the resulting template deploys? You can add the integ test to this folder https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk-testing/framework-integ/test/aws-cloudformation/test

**Comment by bertrandda** [comment]
> @bertrandda in order to use language extensions don't you need to include the transform in the template?
> 
> Can you also add an integration test to make sure that the resulting template deploys? You can add the integ test to this folder https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk-testing/framework-integ/test/aws-cloudformation/test

I added an integ tests with a basic template

File: packages/@aws-cdk-testing/framework-integ/test/aws-cloudformation/test/integ.import-template.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as core from 'aws-cdk-lib';
+import * as inc from 'aws-cdk-lib/cloudformation-include';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+
+const app = new core.App();
+
+const stack = new core.Stack(app, 'ImportStack');
+
+new inc.CfnInclude(stack, 'ImportStack', {
```
Comment (corymhall) [inline packages/@aws-cdk-testing/framework-integ/test/aws-cloudformation/test/integ.import-template.ts:9]: We need a test that doesn't include `CfnInclude`. We also want to test that cdk will synth and deploy a valid template.

**Review by corymhall — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk-testing/framework-integ/test/aws-cloudformation/test/integ.import-template.ts (hunk 1 above)
Comment (bertrandda) [reply packages/@aws-cdk-testing/framework-integ/test/aws-cloudformation/test/integ.import-template.ts:9]: I am not sure what do you expected. Just a basic stack creation (with for example a sqs queue)?

**Pull request closed without merging**